Mathlib/AlgebraicGeometry/EllipticCurve/Jacobian.lean

```diff
@@ -489,22 +498,39 @@ lemma Y_eq_negY_of_Y_eq [NoZeroDivisors R] {P Q : Fin 3 → R} (hQz : Q z ≠ 0)
   mul_left_injective₀ (pow_ne_zero 3 hQz) <| by
     linear_combination (norm := ring1) -Y_sub_Y_add_Y_sub_negY P Q hx + hy + hy'
 
-lemma nonsingular_iff_of_Y_eq_negY {P : Fin 3 → F} (hPy : P y = W.negY P) (hPz : P z ≠ 0) :
-    W.Nonsingular P ↔ W.Equation P ∧ eval P W.polynomialX ≠ 0 := by
-  rw [nonsingular_iff_of_Z_ne_zero hPz, ← Y_ne_negY_iff, hPy, ne_self_iff_false, or_false]
-
 end Negation
 
 section Doubling
 
 /-! ### Doubling formulae -/
 
+variable (W') in
+/-- The unit associated to the doubling of a 2-torsion point.
+More specifically, the unit `u` such that `W.add P P = u • ![1, 1, 0]` where `P = W.neg P`. -/
+noncomputable def dblU (P : Fin 3 → R) : R :=
+  eval P W'.polynomialX
```

Comment:
I [once remarked](https://github.com/leanprover-community/mathlib4/pull/9405#pullrequestreview-2072733853) that this expression appears in `dblX` and `dblY` as well. Since you are now introducing a shorthand for it, maybe use it there as well?

Reply:
Sure, why not.